web/api/v1/api_test.go

```diff
@@ -1466,9 +1466,12 @@ func testEndpoints(t *testing.T, api *API, tr *testTargetRetriever, testLabelAPI
 		if m == http.MethodPost {
 			r, err := http.NewRequest(m, "http://example.com", strings.NewReader(q.Encode()))
 			r.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+			r.RemoteAddr = "127.0.0.1:20201"
```

Comment:
Where is this tested?

Reply:
It is not tested but tests fail without this. Not sure how to test this as it is passed to the context in a private type that is not usable in this package (only usable in promql).

Reply:
Why do they fail?

Reply:
We fail the api call if net.SplitHostPort(r.RemoteAddr) fails.